ethcore/wasm/src/runtime.rs

```diff
@@ -161,23 +161,17 @@ impl<'a> PendingTransaction for Runtime<'a> {
 		}
 
 		match call_result {
-			MessageCallResult::Success(gas_left, return_data)
-			| MessageCallResult::Reverted(gas_left, return_data) => {
-				// cannot overflow, before making call gas_counter was incremented with gas, and gas_left < gas
```

Comment:
This comment seems to be claiming that the `gas_counter` has already been incremented with the previous `gas_left`. And the code that follows is refunding the unused gas that was charged for this call. Does `ext.call` charge gas?

Reply:
> This comment seems to be claiming that the `gas_counter` has already been incremented with the previous `gas_left`

I don't see that happening anywhere. If you look at the arguments to the invocation of `self.ext.call`, there's no mutable reference to `self` that would provide `Externalities` (or the downstream `Executive` with access to `self.gas_counter`.

Perhaps what we're seeing is an artifact of removing gas as a parameter provided by the caller by providing the callee all the remaining gas. In the old world, one could imagine incrementing 	`self.gas_counter` with all of the user-provided gas and then refunding. In this world, the remaining gas is simply the gas that wasn't used by the subcall.

Reply:
Do we have an argument for why we removed as as a parameter or was it just a decision based on convenience/expediency?

Reply:
The idea of a gas limit for sub calls implies that you don’t trust the called code to do the right thing. You shouldn’t be calling untrusted code. Or, at least, I can’t think of any good reasons.